desc/objectives/objective_funs.py

```diff
@@ -166,11 +166,14 @@ def callback(self, x):
             State vector.
 
         """
-        f = self.compute_scalar(x)
+        if self.compiled and self._compile_mode in {"scalar", "all"}:
+            f = self.compute_scalar(x)
+        else:
+            f = jnp.sum(self.compute(x) ** 2) / 2
         print("Total (sum of squares): {:10.3e}, ".format(f))
         kwargs = self.unpack_state(x)
         for obj in self.objectives:
```

Comment:
@ddudt this is the issue mentioned above. If we want to print the values of the individual objectives, there's always going to be some overhead since we only ever compile the combined compute function

Reply:
Sort of. Whether you jit it and use it once or don't jit at all, either way that function eval takes a relatively long time (in some cases one time jit is faster than no jit) since we don't use it anywhere else so can't amortize the cost. It's not a ton of overhead, maybe a few seconds. But depends on how detailed we want the printout to be if there are multiple objectives

Reply:
hm ok, I would think it would be useful to have both the overall objective value and the individual ones, so you can see if maybe the overall objective decrease was due only to like one sub-objective decreasing.

Though if there are a lot of sub-objectives maybe this would be less useful

Reply:
I think it's fine to leave it as is for now. If we are really concerned about computation speed we can run with `verbose=0` to suppress the callback output anyways.